File: app/src/main/java/im/tox/antox/callbacks/AntoxOnStatusMessageCallback.java
package im.tox.antox.callbacks;

import android.content.Context;
import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;

import im.tox.antox.data.AntoxDB;
import im.tox.antox.tox.ToxSingleton;
import im.tox.antox.utils.AntoxFriend;
import im.tox.antox.utils.Constants;
import im.tox.jtoxcore.callbacks.OnStatusMessageCallback;

public class AntoxOnStatusMessageCallback implements OnStatusMessageCallback<AntoxFriend> {

    private final static String TAG = "im.tox.antox.TAG";
    private Context ctx;

    ToxSingleton toxSingleton = ToxSingleton.getInstance();
    public AntoxOnStatusMessageCallback(Context ctx) {
        this.ctx = ctx;
    }

    @Override
    public void execute(AntoxFriend friend, String newStatus) {
        AntoxDB db = new AntoxDB(ctx);
        db.updateStatusMessage(friend.getId(), newStatus);
        db.close();
        toxSingleton.updateFriendsList(ctx);
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnStatusMessageCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import im.tox.antox.data.{State, AntoxDB}
import im.tox.antox.tox.ToxSingleton
import im.tox.antox.utils.UIUtils
import im.tox.tox4j.core.callbacks.FriendStatusMessageCallback

object AntoxOnStatusMessageCallback {

  private val TAG = "im.tox.antox.TAG"
}

class AntoxOnStatusMessageCallback(private var ctx: Context) extends FriendStatusMessageCallback {

  override def friendStatusMessage(friendNumber: Int, messageBytes: Array[Byte]): Unit = {
    val statusMessage = UIUtils.removeNewlines(new String(messageBytes, "UTF-8"))

    ToxSingleton.getAntoxFriend(friendNumber).get.setStatusMessage(statusMessage)

    val db = State.db
    db.updateContactStatusMessage(ToxSingleton.tox.getFriendKey(friendNumber), statusMessage)
  }
}
